edweiss-app/components/todo/abstractTodoEditor.tsx

```diff
@@ -0,0 +1,314 @@
+/**
+ * @file abstractTodoEditor.tsx
+ * @description Module for editing todo items in the edweiss app
+ * @author Adamm Alaoui
+ */
+
+// ------------------------------------------------------------
+// --------------- Import Modules & Components ----------------
+// ------------------------------------------------------------
+
+import TScrollView from '@/components/core/containers/TScrollView';
+import TTouchableOpacity from '@/components/core/containers/TTouchableOpacity';
+import TView from '@/components/core/containers/TView';
+import RouteHeader from '@/components/core/header/RouteHeader';
+import Icon from '@/components/core/Icon';
+import TText from '@/components/core/TText';
+import FancyButton from '@/components/input/FancyButton';
+import FancyTextInput from '@/components/input/FancyTextInput';
+import { callFunction } from '@/config/firebase';
+import t from '@/config/i18config';
+import { LightDarkProps } from '@/constants/Colors';
+import Todolist from '@/model/todo';
+import { Time } from '@/utils/time';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import { router } from 'expo-router';
+import { useState } from 'react';
+import Toast from 'react-native-toast-message';
+import { sameTodos } from '../../utils/todo/utilsFunctions';
+import { StatusChanger } from './todoDisplay';
+import Functions = Todolist.Functions;
+
+// Types
+type TodoStatus = Todolist.TodoStatus;
+
+
+// ------------------------------------------------------------
+// --------------  The Custumisable Todo Editor   -------------
+// ------------------------------------------------------------
+
+export const AbstractTodoEditor: React.FC<{
+    editable?: boolean;
+    todo?: Todolist.Todo;
+    id?: string;
+
+} & LightDarkProps
+> = ({ editable = false, todo, id }) => {
+
+    const providedDate = editable && todo && todo.dueDate;
+
+    // Use states
+    const [name, setName] = useState(editable ? todo!.name : "");
+    const [status, setStatus] = useState<TodoStatus>(editable ? todo!.status : "yet");
+    const [description, setDescription] = useState(editable ? todo!.description : "");
+    const [date, setDate] = useState((providedDate) ? Time.toDate(todo.dueDate!) : new Date());
```

Comment:
You can use just one useState for Date, for better lisibility

Reply:
Yes I aggree as this is a lot of hooks. In earlier versions, I have been using a newTodo hook. However, as I refractaor the cloud function to overcome the Date issues that I mentionned at the top, I had no choice but to put a hook for each field of the Todo.